sympy/integrals/manualintegrate.py

```diff
@@ -2003,6 +2030,42 @@ def rewrites_rule(integral):
 def fallback_rule(integral):
     return DontKnowRule(*integral)
 
+def has_cmplx_exp_pair(expr):
+    """
+    Check if the expression contains a pair of complex exponentials
+    that can be simplified to a sin or cos function.
+    """
+    exp_args = set()
+
+    for sub in preorder_traversal(expr):
+        if sub.func == exp:
+            arg = simplify(sub.args[0])
```

Comment:
The `simplify` function should not be used in library code.

Reply:
Why? It is already used all over the manualintegrate module.
What alternative should I use? I am using it to normalize expressions to check if they are the same or not. So `exp(-x)+exp(-(-x))` is detected as a valid match.

Reply:
The simplify function is slow and heuristic. It leads to unpredictable outputs and it slows everything down.

If there is a need for some simplification in library code then you should figure out which specific simplification function is needed rather than calling simplify which heuristically tries many of them.

Reply:
Makes sense. I will apply some steps like expand, factor terms, powsimp; and see where that is going.